File: PharmaSW/src/main/java/br/com/pharmasw/api/modelo/Pedido.java
package br.com.pharmasw.api.modelo;

import br.com.pharmasw.api.modelo.enums.StatusPedido;
import jakarta.persistence.*;

import java.time.LocalDate;
import java.util.Date;
import java.util.List;

@Entity(name = "pedidos")
@Table(name = "pedidos")
public class Pedido {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "endereco_id")
    private Endereco endereco;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "metodo_pagamento_id")
    private MetodosPagamento metodosPagamento;

    @Enumerated(EnumType.STRING)
    private StatusPedido statusPedido;

    private LocalDate dataCompra;
    private Double subTotal;
    private Double total;
    private Double frete;

    @ElementCollection
    private List<ItemPedido> itemsPedido;

    @Transient
    private Long idCliente;
    @Transient
    private Long idEndereco;
    @Transient
    private Long idMetodoPagamento;


    public Pedido() {

    }


    public Long getId() {
        return id;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public Endereco getEndereco() {
        return endereco;
    }

    public void setEndereco(Endereco endereco) {
        this.endereco = endereco;
    }

    public MetodosPagamento getMetodosPagamento() {
        return metodosPagamento;
    }

    public void setMetodosPagamento(MetodosPagamento metodosPagamento) {
        this.metodosPagamento = metodosPagamento;
    }

    public LocalDate getDataCompra() {
        return dataCompra;
    }

    public void setDataCompra(LocalDate dataCompra) {
        this.dataCompra = dataCompra;
    }

    public StatusPedido getStatusPedido() {
        return statusPedido;
    }

    public void setStatusPedido(StatusPedido statusPedido) {
        this.statusPedido = statusPedido;
    }

    public Double getSubTotal() {
        return subTotal;
    }

    public void setSubTotal(Double subTotal) {
        this.subTotal = subTotal;
    }

    public Double getTotal() {
        return total;
    }

    public void setTotal(Double total) {
        this.total = total;
    }

    public Double getFrete() {
        return frete;
    }

    public void setFrete(Double frete) {
        this.frete = frete;
    }

    public List<ItemPedido> getItemsPedido() {
        return itemsPedido;
    }

    public void setItemsPedido(List<ItemPedido> itemsPedido) {
        this.itemsPedido = itemsPedido;
    }

    public Long getIdCliente() {
        return idCliente;
    }

    public void setIdCliente(Long idCliente) {
        this.idCliente = idCliente;
    }

    public Long getIdEndereco() {
        return idEndereco;
    }

    public void setIdEndereco(Long idEndereco) {
        this.idEndereco = idEndereco;
    }

    public Long getIdMetodoPagamento() {
        return idMetodoPagamento;
    }

    public void setIdMetodoPagamento(Long idMetodoPagamento) {
        this.idMetodoPagamento = idMetodoPagamento;
    }
}

File: pharmasw-front/src/app/modelo/Pedido.ts
import { Cliente } from './Cliente';
import { Endereco } from "./Endereco";
import { StatusPedido } from "./enums/StatusPedido";
import { ItemPedido } from "./ItemPedido";
import { MetodosPagamento } from "./MetodosPagamento";

export class Pedido {
  id: number;
  idCliente?: number;
  idEndereco?: number;
  idMetodoPagamento?: number;
  cliente: Cliente;
  endereco: Endereco;
  metodosPagamento : MetodosPagamento;
  statusPedido: StatusPedido;
  dataCompra: Date;
  subTotal: number;
  total: number;
  frete: number;
  itemsPedido: ItemPedido[];

  constructor(
    idCliente: number = 0,
    idEndereco: number = 0,
    idMetodoPagamento: number = 0,
    subTotal: number,
    total: number,
    frete: number,
    itemsPedido: ItemPedido[]

  ) {
    this.idCliente = idCliente;
    this.idEndereco = idEndereco;
    this.idMetodoPagamento = idMetodoPagamento;
    this.subTotal = subTotal;
    this.total = total;
    this.frete = frete;
    this.itemsPedido = itemsPedido;
  }
}
